os/src/Path.scala

```diff
@@ -398,12 +400,18 @@ object Path {
 
   def apply[T: PathConvertible](f: T, base: Path): Path = apply(FilePath(f), base)
   def apply[T: PathConvertible](f0: T): Path = {
-    val f = implicitly[PathConvertible[T]].apply(f0)
-    if (f.iterator.asScala.count(_.startsWith("..")) > f.getNameCount / 2) {
-      throw PathError.AbsolutePathOutsideRoot
+    // drive letter prefix is empty unless running in Windows.
+    val normalized = {
+      val f = if (rootRelative(f0)) {
+        Paths.get(s"$platformPrefix$f0")
+      } else {
+        implicitly[PathConvertible[T]].apply(f0)
+      }
+      if (f.iterator.asScala.count(_.startsWith("..")) > f.getNameCount / 2) {
+        throw PathError.AbsolutePathOutsideRoot
+      }
```

Comment:
We can re-arrange this to move the `throw` outside of the `val normalized`? All these are method-local `val`s anyway, so there's no visibility/privacy concerns, and it would be nice to remove a layer of nesting

Reply:
done.